Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2406,6 +2406,10 @@ extension BrowserViewController: ReaderModeBarViewDelegate {
 
 extension BrowserViewController: IntroViewControllerDelegate {
     @discardableResult func presentIntroViewController(_ force: Bool = false) -> Bool {
+        if let deeplink = self.profile.prefs.stringForKey("AdjustDeeplinkKey"), let url = URL(string: deeplink) {
+            self.launchFxAFromDeeplinkURL(url)
```

Comment:
`launchFxAFromDeeplinkURL` pops open a VC. and `presentIntroViewController` also pops up a VC. You don't seem to be stopping the controlflow?
I think you need a return after the `self.launchFxAFromDeeplinkURL`